lib/Twig/Template.php

```diff
@@ -607,15 +618,23 @@ protected function getAttribute($object, $item, array $arguments = array(), $typ
             $this->env->getExtension('Twig_Extension_Sandbox')->checkMethodAllowed($object, $method);
         }
 
-        // Some objects throw exceptions when they have __call, and the method we try
-        // to call is not supported. If ignoreStrictCheck is true, we should return null.
-        try {
-            $ret = call_user_func_array(array($object, $method), $arguments);
-        } catch (BadMethodCallException $e) {
-            if ($call && ($ignoreStrictCheck || !$this->env->isStrictVariables())) {
-                return;
+        if ($call) {
+            // Some objects throw exceptions when they have __call, and the method we try
+            // to call is not supported. If ignoreStrictCheck is true, we should return null.
+            try {
+                $ret = $object->__call($method, $arguments);
+            } catch (BadMethodCallException $e) {
+                if ($ignoreStrictCheck || !$this->env->isStrictVariables()) {
+                    return;
+                }
+                throw $e;
             }
-            throw $e;
+        } elseif (!$arguments) {
+            $ret = $object->$method();
+        } elseif (PHP_VERSION_ID >= 50600) {
+            $ret = twig_call_method($object, $method, $arguments);
```

Comment:
optim 5 bis: use argument unpacking in a wrapper.
As surprising as this seems, its faster to call this wrapper that using `call_user_func_array` directly. Should be related to the extra array allocation that it requires.
A micro-benchm shows this on my PHP 7.1:
- x1.0: call_user_func_array(reference)
- x1.8: switch (but unsafe as spotted by @stof 
- x2.3: twig_call_method
- x3.5: unpacking (but Twig's min version is PHP 5.2, not 5.6)